examples/decomposition/plot_faces_decomposition.py

```diff
@@ -122,7 +123,8 @@ def plot_gallery(title, images, n_col=n_col, n_row=n_row):
         components_ = estimator.cluster_centers_
     else:
         components_ = estimator.components_
-    if hasattr(estimator, 'noise_variance_'):
+    if (hasattr(estimator, 'noise_variance_') and
+            estimator.noise_variance_.shape != ()):
```

Comment:
The reason for this is that `.noise_variance_` is a `numpy.float64` for `PCA`. Maybe it is slightly simpler to do `estimator.noise_variance_.ndim > 0` and potentially add a comment to explain why we need the additional check.